catalog/consul.go

```diff
@@ -306,11 +319,13 @@ func (c *consul) create(services map[string]service) int {
 					meta[ConsulAWSNS] = ns
 					meta[ConsulAWSID] = n.awsID
 					service := api.AgentService{
-						ID:      id,
-						Service: name,
-						Tags:    []string{ConsulAWSTag},
-						Address: h,
-						Meta:    meta,
+						ID:        id,
+						Service:   name,
+						Tags:      []string{ConsulAWSTag},
+						Address:   h,
+						Meta:      meta,
+						Namespace: c.namespace,
+						Partition: c.adminPartition,
```

Comment:
I couldn't find api docs for this field 🤔, maybe I am missing something? But this is definitely a valid field and is probably working fine: https://github.com/hashicorp/consul/blob/main/api/agent.go.